### PR TITLE
[JdbcIO] - Adding option for max batch buffering duration

### PR DESCRIPTION
**Pull request by scott-strong**
Currently the max buffering duration for batch processing during JDBC writes is not configurable and defaulted to 200ms. I'm proposing that we add an option to configure this setting to allow for a longer duration period to accumulate batched elements. We've run into an issue where batches are committed too frequently causing many small batches to be written rather than fewer larger batches, and I'm hoping to tweak this setting to allow for the latter.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by scott-strong** [comment]
test failures here seem unrelated but I can't re-run, going to assign reviewers

**Comment by scott-strong** [comment]
assign set of reviewers

**Comment by Abacn** [comment]
The idea sounds good to me.

There are two options controlling the batch, batch_size and buffering_duration.

For example, if batch_size set to 100 and buffering_duration=200ms.

- if batch size reaches 1000 within 100ms
- if batch size does not reach 1000 within 200ms

If I understood correctly, in both cases, the pipeline will commit the buffer? Good to add some documentation about the spec.

**Comment by scott-strong** [comment]
@Abacn yes exactly, so currently we can increase the batch size but not the max buffering duration. Each is a way to configure when the pipeline will commit the buffer. I can add documentation about this.

**Comment by scott-strong** [comment]
@Abacn I added more to the javadoc for both `withBatchSize` and `withMaxBatchBufferingDuration`, let me know if it sounds good to you

**Comment by Abacn** [comment]
Thanks. For the Spotless (formatting) check failure, please run `./gradlew :sdks:java:io:jdbc:spotlessApply` then push the changes

File: sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java [hunk 1]
```diff
@@ -1971,6 +1982,7 @@ public PCollection<V> expand(PCollection<T> input) {
                       .setRetryConfiguration(getRetryConfiguration())
                       .setReturnResults(true)
                       .setBatchSize(1L)
+                      .setMaxBatchBufferingDuration(200L)
```
Comment (Abacn) [inline sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java:1985]: Use the constant (DEFAULT_MAX_BATCH_BUFFERING_DURATION) as it is defined there

**Review by Abacn — COMMENTED** [review]
Thanks for the changes. Just had one comment

File: sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java (hunk 1 above)
Comment (scott-strong) [reply sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java]: thanks, just made that change

**Review by Abacn — APPROVED** [review]
Thank you!

**Pull request merged**